sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -1111,7 +1129,12 @@ private[hive] object HiveClientImpl extends Logging {
     //   struct<x:int,y.z:int> -> struct<`x`:int,`y.z`:int>
     //   array<struct<x:int,y.z:int>> -> array<struct<`x`:int,`y.z`:int>>
     //   map<string,struct<x:int,y.z:int>> -> map<string,struct<`x`:int,`y.z`:int>>
-    val typeStr = hc.getType.replaceAll("(?<=struct<|,)([^,<:]+)(?=:)", "`$1`")
+    val typeStr = if (SQLConf.get.getConf(QUOTE_HIVE_STRUCT_FIELD_NAME) &&
+        hc.getType.indexOf('`') < 0) { // This a defensive code for possible changes in HMS
```

Comment:
Do we really need this code?

- For old Spark versions, they will not be able to write `typeStr` not supported by `CatalystSqlParser.parseDataType` into HMS.
- For new Spark versions, including this change, this case should be taken care of. Right?

Reply:
Thanks, @zsxwing. Just as the same answer to the similar question that @cloud-fan raised before, this is just a defensive code for possible changes in HMS